No field mDefaultJSExceptionHandler in class Lcom/facebook/react/devsupport/u

### Minimal reproducible example

Will add if needed

### What platform(s) does this occur on?

Android

### Where did you reproduce the issue?

in a standalone app

### Summary

@alanjhughes Running into this on RN `0.75.2`, `0.74.5` works fine. Looks like the error bubbles up from `expo.modules.updates`.

Using expo 51.0.30 with old arch and proguard (I've had to manually add proguard rules for unistyles to keep classes, I can see [here](https://github.com/expo/expo/blob/main/packages/expo-updates/android/proguard-rules.pro) that the same missing variable has a proguard rule to keep it. Maybe its a proguard issue? idk  ). This only happens in release builds on Android.

Logcat
```bash
2024-08-22 20:53:21.679 10663-10663 SearchResultIcon        pid-10663                            D  Launching Omnid icon with component ComponentInfo{com.cupoc.omnid/com.cupoc.omnid.MainActivity}
2024-08-22 20:53:21.697  1724-27338 ActivityTaskManager     pid-1724                             I  START u0 {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=com.cupoc.omnid/.MainActivity bnds=[53,300][248,525]} with LAUNCH_SINGLE_TASK from uid 10227 (BAL_ALLOW_ALLOWLISTED_COMPONENT) result code=0
2024-08-22 20:53:21.698  2540-2579  WindowManagerShell      pid-2540                             V  Transition requested (#15063): android.os.BinderProxy@93dc852 TransitionRequestInfo { type = OPEN, triggerTask = TaskInfo{userId=0 taskId=10691 displayId=0 isRunning=true baseIntent=Intent { act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=com.cupoc.omnid/.MainActivity } baseActivity=ComponentInfo{com.cupoc.omnid/com.cupoc.omnid.MainActivity} topActivity=ComponentInfo{com.cupoc.omnid/com.cupoc.omnid.MainActivity} origActivity=null realActivity=ComponentInfo{com.cupoc.omnid/com.cupoc.omnid.MainActivity} numActivities=1 lastActiveTime=2860245345 supportsMultiWindow=true resizeMode=1 isResizeable=true minWidth=-1 minHeight=-1 defaultMinSize=220 token=WCT{android.window.IWindowContainerToken$Stub$Proxy@3c37723} topActivityType=1 pictureInPictureParams=null shouldDockBigOverlays=false launchIntoPipHostTaskId=-1 lastParentTaskIdBeforePip=-1 displayCutoutSafeInsets=Rect(0, 132 - 0, 0) topActivityInfo=ActivityInfo{989420 com.cupoc.omnid.MainActivity} launchCookies=[android.os.BinderProxy@ed901d9] positionInParent=Point(0, 0) parentTaskId=-1 isFocused=false isVisible=false isVisibleRequested=false isSleeping=false locusId=null displayAreaFeatureId=1 isTopActivityTransparent=false appCompatTaskInfo=AppCompatTaskInfo { topActivityInSizeCompat=false topActivityEligibleForLetterboxEducation= false isLetterboxDoubleTapEnabled= false topActivityEligibleForUserAspectRatioButton= false topActivityBoundsLetterboxed= false isFromLetterboxDoubleTap= false topActivityLetterboxVerticalPosition= -1 topActivityLetterboxHorizontalPosition= -1 topActivityLetterboxWidth=1080 topActivityLetterboxHeight=2400 isUserFullscreenOverrideEnabled=false isSystemFullscreenOverrideEnabled=false cameraCompatControlState=hidden}}, pipTask = null, remoteTransition = RemoteTransition { remoteTransition = android.window.IRemoteTransition$Stub$Proxy@7cc179e, appThread = android.app.IApplicationThread$Stub$Proxy@34c627f, debugName = QuickstepLaunch }, displayChange = null, flags = 0, debugId = 15063 }
2024-08-22 20:53:21.699 10663-10663 SearchSessionManager    pid-10663                            D  notifyEvent: query=o at 2860230277 Bundle{timeout=200 entry=2 } targetId=APPLICATION:0:com.cupoc.omnid:com.cupoc.omnid.MainActivity eventType=5
2024-08-22 20:53:21.713  1724-18149 CoreBackPreview         pid-1724                             D  Window{6fd7da1 u0 Splash Screen com.cupoc.omnid}: Setting back callback OnBackInvokedCallbackInfo{mCallback=android.window.IOnBackInvokedCallback$Stub$Proxy@dbc0523, mPriority=0, mIsAnimationCallback=false}
2024-08-22 20:53:21.715   710-710   gralloc4                pid-710                              W  Unable to set buffer name VRI[omnid]#18289(BLAST Consumer)18289: File name too long
2024-08-22 20:53:21.716   710-710   gralloc4                pid-710                              W  Unable to set buffer name VRI[omnid]#18289(BLAST Consumer)18289: File name too long
2024-08-22 20:53:21.717  1724-1790  ActivityManager         pid-1724                             I  Start proc 12057:com.cupoc.omnid/u0a516 for next-top-activity {com.cupoc.omnid/com.cupoc.omnid.MainActivity}
2024-08-22 20:53:21.717   710-710   gralloc4                pid-710                              W  Unable to set buffer name VRI[omnid]#18289(BLAST Consumer)18289: File name too long
2024-08-22 20:53:21.718   710-710   gralloc4                pid-710                              W  Unable to set buffer name VRI[omnid]#18289(BLAST Consumer)18289: File name too long
2024-08-22 20:53:21.720  1724-1766  WindowManager           pid-1724                             V  Sent Transition (#15063) createdAt=08-22 20:53:21.688 via request=TransitionRequestInfo { type = OPEN, triggerTask = TaskInfo{userId=0 taskId=10691 displayId=0 isRunning=true baseIntent=Intent { act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=com.cupoc.omnid/.MainActivity } baseActivity=ComponentInfo{com.cupoc.omnid/com.cupoc.omnid.MainActivity} topActivity=ComponentInfo{com.cupoc.omnid/com.cupoc.omnid.MainActivity} origActivity=null realActivity=ComponentInfo{com.cupoc.omnid/com.cupoc.omnid.MainActivity} numActivities=1 lastActiveTime=2860245345 supportsMultiWindow=true resizeMode=1 isResizeable=true minWidth=-1 minHeight=-1 defaultMinSize=220 token=WCT{RemoteToken{f3fd07f Task{38bea69 #10691 type=standard A=10516:com.cupoc.omnid}}} topActivityType=1 pictureInPictureParams=null shouldDockBigOverlays=false launchIntoPipHostTaskId=-1 lastParentTaskIdBeforePip=-1 displayCutoutSafeInsets=Rect(0, 132 - 0, 0) topActivityInfo=ActivityInfo{6ba84f0 com.cupoc.omnid.MainActivity} launchCookies=[android.os.BinderProxy@6d30095] positionInParent=Point(0, 0) parentTaskId=-1 isFocused=false isVisible=false isVisibleRequested=false isSleeping=false locusId=null displayAreaFeatureId=1 isTopActivityTransparent=false appCompatTaskInfo=AppCompatTaskInfo { topActivityInSizeCompat=false topActivityEligibleForLetterboxEducation= false isLetterboxDoubleTapEnabled= false topActivityEligibleForUserAspectRatioButton= false topActivityBoundsLetterboxed= false isFromLetterboxDoubleTap= false topActivityLetterboxVerticalPosition= -1 topActivityLetterboxHorizontalPosition= -1 topActivityLetterboxWidth=1080 topActivityLetterboxHeight=2400 isUserFullscreenOverrideEnabled=false isSystemFullscreenOverrideEnabled=false cameraCompatControlState=hidden}}, pipTask = null, remoteTransition = RemoteTransition { remoteTransition = android.window.IRemoteTransition$Stub$Proxy@a42d5aa, appThread = android.app.IApplicationThread$Stub$Proxy@70af59b, debugName = QuickstepLaunch }, displayChange = null, flags = 0, debugId = 15063 }
2024-08-22 20:53:21.720  1724-1766  WindowManager           pid-1724                             V      info={id=15063 t=OPEN f=0x0 trk=0 r=[0@Point(0, 0)] c=[
                                                                                                            {WCT{RemoteToken{f3fd07f Task{38bea69 #10691 type=standard A=10516:com.cupoc.omnid}}} m=OPEN f=NONE leash=Surface(name=Task=10691)/@0x7373fd9 sb=Rect(0, 0 - 1080, 2400) eb=Rect(0, 0 - 1080, 2400) d=0},
                                                                                                            {WCT{RemoteToken{7fc124a Task{3e69cc6 #1 type=undefined}}} m=TO_BACK f=SHOW_WALLPAPER leash=Surface(name=Task=1)/@0x83ded21 sb=Rect(0, 0 - 1080, 2400) eb=Rect(0, 0 - 1080, 2400) d=0}
                                                                                                        ]}
2024-08-22 20:53:21.720   710-710   gralloc4                pid-710                              W  Unable to set buffer name VRI[omnid]#18289(BLAST Consumer)18289: File name too long
2024-08-22 20:53:21.724 12057-12057 com.cupoc.omnid         pid-12057                            I  Using CollectorTypeCMC GC.
2024-08-22 20:53:21.728 12057-12057 com.cupoc.omnid         pid-12057                            E  Not starting debugger since process cannot load the jdwp agent.
2024-08-22 20:53:21.740 12057-12057 nativeloader            pid-12057                            D  Configuring clns-6 for other apk /system_ext/framework/com.google.android.camerax.extensions.jar. target_sdk_version=34, uses_libraries=ALL, library_path=/data/app/~~9vzzuPxKNyjSO0uoZjXI2A==/com.cupoc.omnid-KXZ6zW0M5U6g_gw34gdo-A==/lib/arm64:/data/app/~~9vzzuPxKNyjSO0uoZjXI2A==/com.cupoc.omnid-KXZ6zW0M5U6g_gw34gdo-A==/base.apk!/lib/arm64-v8a, permitted_path=/data:/mnt/expand:/data/user/0/com.cupoc.omnid
2024-08-22 20:53:21.741 12057-12057 nativeloader            pid-12057                            D  Configuring clns-7 for other apk /data/app/~~9vzzuPxKNyjSO0uoZjXI2A==/com.cupoc.omnid-KXZ6zW0M5U6g_gw34gdo-A==/base.apk. target_sdk_version=34, uses_libraries=, library_path=/data/app/~~9vzzuPxKNyjSO0uoZjXI2A==/com.cupoc.omnid-KXZ6zW0M5U6g_gw34gdo-A==/lib/arm64:/data/app/~~9vzzuPxKNyjSO0uoZjXI2A==/com.cupoc.omnid-KXZ6zW0M5U6g_gw34gdo-A==/base.apk!/lib/arm64-v8a, permitted_path=/data:/mnt/expand:/data/user/0/com.cupoc.omnid
2024-08-22 20:53:21.745 12057-12057 GraphicsEnvironment     pid-12057                            V  com.cupoc.omnid is not listed in per-application setting
2024-08-22 20:53:21.762 12057-12057 SoLoader                pid-12057                            W  Recording new base apk path: /data/app/~~9vzzuPxKNyjSO0uoZjXI2A==/com.cupoc.omnid-KXZ6zW0M5U6g_gw34gdo-A==/base.apk
                                                                                                    Previously recorded 0 base apk paths.
2024-08-22 20:53:21.773 12057-12086 vulkan                  pid-12057                            D  searching for layers in '/data/app/~~9vzzuPxKNyjSO0uoZjXI2A==/com.cupoc.omnid-KXZ6zW0M5U6g_gw34gdo-A==/lib/arm64'
2024-08-22 20:53:21.773 12057-12086 vulkan                  pid-12057                            D  searching for layers in '/data/app/~~9vzzuPxKNyjSO0uoZjXI2A==/com.cupoc.omnid-KXZ6zW0M5U6g_gw34gdo-A==/base.apk!/lib/arm64-v8a'
2024-08-22 20:53:21.818  1724-27338 CoreBackPreview         pid-1724                             D  Window{ec94567 u0 com.cupoc.omnid/com.cupoc.omnid.MainActivity}: Setting back callback OnBackInvokedCallbackInfo{mCallback=android.window.IOnBackInvokedCallback$Stub$Proxy@1ec1880, mPriority=0, mIsAnimationCallback=false}
2024-08-22 20:53:21.824 12057-12100 dev.expo.updates        pid-12057                            I  {"timestamp":1724340201824,"message":"embeddedAssetFileMap: 778ffc9fe8773a878e9c30a6304784de,png => file:\/\/\/data\/user\/0\/com.cupoc.omnid\/files\/.expo-internal\/778ffc9fe8773a878e9c30a6304784de.png","code":"None","level":"info"}
2024-08-22 20:53:21.824 12057-12100 dev.expo.updates        pid-12057                            I  {"timestamp":1724340201824,"message":"embeddedAssetFileMap: 376d6a4c7f622917c39feb23671ef71d,png => file:\/\/\/data\/user\/0\/com.cupoc.omnid\/files\/.expo-internal\/376d6a4c7f622917c39feb23671ef71d.png","code":"None","level":"info"}
2024-08-22 20:53:21.824 12057-12100 dev.expo.updates        pid-12057                            I  {"timestamp":1724340201824,"message":"embeddedAssetFileMap: c79c3606a1cf168006ad3979763c7e0c,png => file:\/\/\/data\/user\/0\/com.cupoc.omnid\/files\/.expo-internal\/c79c3606a1cf168006ad3979763c7e0c.png","code":"None","level":"info"}
2024-08-22 20:53:21.824 12057-12100 dev.expo.updates        pid-12057                            I  {"timestamp":1724340201824,"message":"embeddedAssetFileMap: 02bc1fa7c0313217bde2d65ccbff40c9,png => file:\/\/\/data\/user\/0\/com.cupoc.omnid\/files\/.expo-internal\/02bc1fa7c0313217bde2d65ccbff40c9.png","code":"None","level":"info"}
2024-08-22 20:53:21.824 12057-12100 dev.expo.updates        pid-12057                            I  {"timestamp":1724340201824,"message":"embeddedAssetFileMap: 35ba0eaec5a4f5ed12ca16fabeae451d,png => file:\/\/\/data\/user\/0\/com.cupoc.omnid\/files\/.expo-internal\/35ba0eaec5a4f5ed12ca16fabeae451d.png","code":"None","level":"info"}
2024-08-22 20:53:21.825 12057-12100 dev.expo.updates        pid-12057                            I  {"timestamp":1724340201825,"message":"embeddedAssetFileMap: 5223c8d9b0d08b82a5670fb5f71faf78,png => file:\/\/\/data\/user\/0\/com.cupoc.omnid\/files\/.expo-internal\/5223c8d9b0d08b82a5670fb5f71faf78.png","code":"None","level":"info"}
2024-08-22 20:53:21.825 12057-12100 dev.expo.updates        pid-12057                            I  {"timestamp":1724340201825,"message":"embeddedAssetFileMap: 563d5e3294b67811d0a1aede6f601e30,png => file:\/\/\/data\/user\/0\/com.cupoc.omnid\/files\/.expo-internal\/563d5e3294b67811d0a1aede6f601e30.png","code":"None","level":"info"}
2024-08-22 20:53:21.825 12057-12100 dev.expo.updates        pid-12057                            I  {"timestamp":1724340201825,"message":"embeddedAssetFileMap: b6c297a501e289394b0bc5dc69c265e6,png => file:\/\/\/data\/user\/0\/com.cupoc.omnid\/files\/.expo-internal\/b6c297a501e289394b0bc5dc69c265e6.png","code":"None","level":"info"}
2024-08-22 20:53:21.825 12057-12100 dev.expo.updates        pid-12057                            I  {"timestamp":1724340201825,"message":"embeddedAssetFileMap: 5974eb3e1c5314e8d5a822702d7d0740,png => file:\/\/\/data\/user\/0\/com.cupoc.omnid\/files\/.expo-internal\/5974eb3e1c5314e8d5a822702d7d0740.png","code":"None","level":"info"}
2024-08-22 20:53:21.825 12057-12100 dev.expo.updates        pid-12057                            I  {"timestamp":1724340201825,"message":"embeddedAssetFileMap: 9d9c5644f55c2f6e4b7f247c378b2fe9,png => file:\/\/\/data\/user\/0\/com.cupoc.omnid\/files\/.expo-internal\/9d9c5644f55c2f6e4b7f247c378b2fe9.png","code":"None","level":"info"}
2024-08-22 20:53:21.825 12057-12100 dev.expo.updates        pid-12057                            I  {"timestamp":1724340201825,"message":"embeddedAssetFileMap: 47f13fcfaf4f2d8bbb3423adc030c12f,zkey => file:\/\/\/data\/user\/0\/com.cupoc.omnid\/files\/.expo-internal\/47f13fcfaf4f2d8bbb3423adc030c12f.zkey","code":"None","level":"info"}
2024-08-22 20:53:21.825 12057-12100 dev.expo.updates        pid-12057                            I  {"timestamp":1724340201825,"message":"embeddedAssetFileMap: 486bd5e1dba1debf9273e6742edf9c9f,wasm => file:\/\/\/data\/user\/0\/com.cupoc.omnid\/files\/.expo-internal\/486bd5e1dba1debf9273e6742edf9c9f.wasm","code":"None","level":"info"}
2024-08-22 20:53:21.848  1724-1766  ActivityTaskManager     pid-1724                             I  Displayed com.cupoc.omnid/.MainActivity for user 0: +161ms
2024-08-22 20:53:21.898 12057-12100 AndroidRuntime          pid-12057                            E  FATAL EXCEPTION: DefaultDispatcher-worker-1
                                                                                                    Process: com.cupoc.omnid, PID: 12057
                                                                                                    java.lang.reflect.InvocationTargetException
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at T6.g.A(SourceFile:43)
                                                                                                    	at T6.g.B(SourceFile:34)
                                                                                                    	at T6.g.t(SourceFile:1)
                                                                                                    	at T6.f.run(SourceFile:1)
                                                                                                    	at expo.modules.updates.UpdatesPackage$c$b.o(SourceFile:13)
                                                                                                    	at I9.a.d(SourceFile:12)
                                                                                                    	at kb.P.run(SourceFile:129)
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:959)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:100)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:232)
                                                                                                    	at android.os.Looper.loop(Looper.java:317)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:8592)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:580)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:878)
                                                                                                    	Suppressed: nb.i: [B0{Cancelling}@bd5db56, Dispatchers.IO]
                                                                                                    Caused by: java.lang.NoSuchFieldException: No field mDefaultJSExceptionHandler in class Lcom/facebook/react/devsupport/u; (declaration of 'com.facebook.react.devsupport.u' appears in /data/app/~~9vzzuPxKNyjSO0uoZjXI2A==/com.cupoc.omnid-KXZ6zW0M5U6g_gw34gdo-A==/base.apk)
                                                                                                    	at java.lang.Class.getDeclaredField(Native Method)
                                                                                                    	at f9.d.n(SourceFile:24)
                                                                                                    	at f9.d.m(SourceFile:9)
                                                                                                    	at f9.d.q(SourceFile:9)
                                                                                                    	at k9.k.E(SourceFile:13)
                                                                                                    	at expo.modules.updates.b.b(SourceFile:8)
                                                                                                    	at expo.modules.updates.UpdatesPackage$d.b(SourceFile:12)
                                                                                                    	at T6.i.createReactInstanceManager(SourceFile:57)
                                                                                                    	at com.facebook.react.M.c(SourceFile:15)
                                                                                                    	at com.facebook.react.x.i(SourceFile:50)
                                                                                                    	at com.facebook.react.t.loadApp(SourceFile:3)
                                                                                                    	... 16 more
2024-08-22 20:53:21.898  1724-2620  ImeTracker              pid-1724                             I  com.cupoc.omnid:94eaffc2: onRequestHide at ORIGIN_SERVER reason HIDE_UNSPECIFIED_WINDOW fromUser false
2024-08-22 20:53:21.902  1724-27338 ActivityTaskManager     pid-1724                             W    Force finishing activity com.cupoc.omnid/.MainActivity
2024-08-22 20:53:21.911 12057-12057 TransactionExecutor     pid-12057                            E  Failed to execute the transaction: tId:1361371967 ClientTransaction{
                                                                                                    tId:1361371967   transactionItems=[
                                                                                                    tId:1361371967     PauseActivityItem{mActivityToken=android.os.BinderProxy@dedfa13,finished=true,userLeaving=false,configChanges=0,dontReport=false,autoEnteringPip=false}
                                                                                                    tId:1361371967     Target activity: com.cupoc.omnid.MainActivity
                                                                                                    tId:1361371967   ]
                                                                                                    tId:1361371967 }
2024-08-22 20:53:21.914 25520-25520 GoogleInpu...hodService pid-25520                            I  GoogleInputMethodService.onStartInput():1982 onStartInput(EditorInfo{EditorInfo{packageName=com.cupoc.omnid, inputType=0, inputTypeString=NULL, enableLearning=false, autoCorrection=false, autoComplete=false, imeOptions=0, privateImeOptions=null, actionName=UNSPECIFIED, actionLabel=null, initialSelStart=-1, initialSelEnd=-1, initialCapsMode=0, label=null, fieldId=0, fieldName=null, extras=null, hintText=null, hintLocales=[]}}, false)
2024-08-22 20:53:21.915  1724-24000 PackageConfigPersister  pid-1724                             W  App-specific configuration not found for packageName: com.cupoc.omnid and userId: 0
2024-08-22 20:53:21.920 10663-10663 ImeTracker              pid-10663                            I  com.cupoc.omnid:94eaffc2: onCancelled at PHASE_CLIENT_ANIMATION_CANCEL
2024-08-22 20:53:21.937  1724-24000 ActivityManager         pid-1724                             I  Process com.cupoc.omnid (pid 12057) has died: prcp TOP 
2024-08-22 20:53:21.937  1724-1745  WindowManager           pid-1724                             I  WIN DEATH: Window{ec94567 u0 com.cupoc.omnid/com.cupoc.omnid.MainActivity}
2024-08-22 20:53:21.937  1724-1745  InputManager-JNI        pid-1724                             W  Input channel object 'ec94567 com.cupoc.omnid/com.cupoc.omnid.MainActivity (client)' was disposed without first being removed with the input manager!
2024-08-22 20:53:21.938  1724-1745  WindowManager           pid-1724                             V  Loading animations: layout params pkg=com.cupoc.omnid resId=0x1030308
2024-08-22 20:53:21.939  1724-18149 ImeTracker              pid-1724                             I  com.cupoc.omnid:49295d04: onRequestHide at ORIGIN_SERVER reason HIDE_REMOVE_CLIENT fromUser false
2024-08-22 20:53:21.939  1724-18149 ImeTracker              pid-1724                             I  com.cupoc.omnid:49295d04: onCancelled at PHASE_SERVER_SHOULD_HIDE
2024-08-22 20:53:21.945  1724-1771  WindowManager           pid-1724                             W  Failed to deliver inset control state change to w=Window{ec94567 u0 com.cupoc.omnid/com.cupoc.omnid.MainActivity EXITING}
                                                                                                    android.os.DeadObjectException
                                                                                                    	at android.os.BinderProxy.transactNative(Native Method)
                                                                                                    	at android.os.BinderProxy.transact(BinderProxy.java:586)
                                                                                                    	at android.view.IWindow$Stub$Proxy.insetsControlChanged(IWindow.java:485)
                                                                                                    	at com.android.server.wm.WindowState.notifyInsetsControlChanged(WindowState.java:3775)
                                                                                                    	at com.android.server.wm.InsetsStateController.lambda$notifyPendingInsetsControlChanged$3(InsetsStateController.java:376)
                                                                                                    	at com.android.server.wm.InsetsStateController.$r8$lambda$MnP7XA1MDcyrUwoj0hcoT5hTO8E(InsetsStateController.java:0)
                                                                                                    	at com.android.server.wm.InsetsStateController$$ExternalSyntheticLambda0.run(R8$$SyntheticClass:0)
                                                                                                    	at com.android.server.wm.WindowAnimator.executeAfterPrepareSurfacesRunnables(WindowAnimator.java:314)
                                                                                                    	at com.android.server.wm.WindowAnimator.animate(WindowAnimator.java:207)
                                                                                                    	at com.android.server.wm.WindowAnimator.lambda$new$1(WindowAnimator.java:99)
                                                                                                    	at com.android.server.wm.WindowAnimator.$r8$lambda$AS_wbK9i-bc6ocCFop7s9PnXP80(WindowAnimator.java:0)
                                                                                                    	at com.android.server.wm.WindowAnimator$$ExternalSyntheticLambda1.doFrame(R8$$SyntheticClass:0)
                                                                                                    	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1404)
                                                                                                    	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1415)
                                                                                                    	at android.view.Choreographer.doCallbacks(Choreographer.java:1015)
                                                                                                    	at android.view.Choreographer.doFrame(Choreographer.java:941)
                                                                                                    	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1389)
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:959)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:100)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:232)
                                                                                                    	at android.os.Looper.loop(Looper.java:317)
                                                                                                    	at android.os.HandlerThread.run(HandlerThread.java:85)
                                                                                                    	at com.android.server.ServiceThread.run(ServiceThread.java:46)
2024-08-22 20:53:21.970  1724-1766  WindowManager           pid-1724                             V      info={id=15064 t=CLOSE f=0x10 trk=0 r=[0@Point(0, 0)] c=[
                                                                                                            {WCT{RemoteToken{7fc124a Task{3e69cc6 #1 type=undefined}}} m=TO_FRONT f=SHOW_WALLPAPER|MOVE_TO_TOP leash=Surface(name=Task=1)/@0x83ded21 sb=Rect(0, 0 - 1080, 2400) eb=Rect(0, 0 - 1080, 2400) d=0},
                                                                                                            {WCT{RemoteToken{f3fd07f Task{38bea69 #10691 type=standard A=10516:com.cupoc.omnid}}} m=CLOSE f=NONE leash=Surface(name=Task=10691)/@0x7373fd9 sb=Rect(0, 0 - 1080, 2400) eb=Rect(0, 0 - 1080, 2400) d=0}
                                                                                                        ]}
2024-08-22 20:53:21.984  1724-1771  WindowManager           pid-1724                             W  Failed to deliver inset control state change to w=Window{ec94567 u0 com.cupoc.omnid/com.cupoc.omnid.MainActivity EXITING}
                                                                                                    android.os.DeadObjectException
                                                                                                    	at android.os.BinderProxy.transactNative(Native Method)
                                                                                                    	at android.os.BinderProxy.transact(BinderProxy.java:586)
                                                                                                    	at android.view.IWindow$Stub$Proxy.insetsControlChanged(IWindow.java:485)
                                                                                                    	at com.android.server.wm.WindowState.notifyInsetsControlChanged(WindowState.java:3775)
                                                                                                    	at com.android.server.wm.InsetsStateController.lambda$notifyPendingInsetsControlChanged$3(InsetsStateController.java:376)
                                                                                                    	at com.android.server.wm.InsetsStateController.$r8$lambda$MnP7XA1MDcyrUwoj0hcoT5hTO8E(InsetsStateController.java:0)
                                                                                                    	at com.android.server.wm.InsetsStateController$$ExternalSyntheticLambda0.run(R8$$SyntheticClass:0)
                                                                                                    	at com.android.server.wm.WindowAnimator.executeAfterPrepareSurfacesRunnables(WindowAnimator.java:314)
                                                                                                    	at com.android.server.wm.WindowAnimator.animate(WindowAnimator.java:207)
                                                                                                    	at com.android.server.wm.WindowAnimator.lambda$new$1(WindowAnimator.java:99)
                                                                                                    	at com.android.server.wm.WindowAnimator.$r8$lambda$AS_wbK9i-bc6ocCFop7s9PnXP80(WindowAnimator.java:0)
                                                                                                    	at com.android.server.wm.WindowAnimator$$ExternalSyntheticLambda1.doFrame(R8$$SyntheticClass:0)
                                                                                                    	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1404)
                                                                                                    	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1415)
                                                                                                    	at android.view.Choreographer.doCallbacks(Choreographer.java:1015)
                                                                                                    	at android.view.Choreographer.doFrame(Choreographer.java:941)
                                                                                                    	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1389)
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:959)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:100)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:232)
                                                                                                    	at android.os.Looper.loop(Looper.java:317)
                                                                                                    	at android.os.HandlerThread.run(HandlerThread.java:85)
                                                                                                    	at com.android.server.ServiceThread.run(ServiceThread.java:46)
2024-08-22 20:53:21.994 10663-10663 HotseatPre...Controller pid-10663                            D  Predicted items: [WorkspaceItemInfo(id=-1 type=APP container=# com.android.launcher3.logger.LauncherAtom$ContainerInfo@19f79dd targetComponent=ComponentInfo{com.cupoc.omnid/com.cupoc.omnid.MainActivity} screen=-1 cell(-1,-1) span(1,1) minSpan(1,1) rank=0 user=UserHandle{0} title=Omnid), WorkspaceItemInfo(id=-1 type=APP container=# com.android.launcher3.logger.LauncherAtom$ContainerInfo@19f79dd targetComponent=ComponentInfo{com.google.android.apps.nbu.files/com.google.android.apps.nbu.files.home.HomeActivity} screen=-1 cell(-1,-1) span(1,1) minSpan(1,1) rank=0 user=UserHandle{0} title=Files), WorkspaceItemInfo(id=-1 type=APP container=# com.android.launcher3.logger.LauncherAtom$ContainerInfo@19f79dd targetComponent=ComponentInfo{slowscript.httpfileserver/slowscript.httpfileserver.MainActivity} screen=-1 cell(-1,-1) span(1,1) minSpan(1,1) rank=0 user=UserHandle{0} title=HTTP File Server), WorkspaceItemInfo(id=-1 type=APP container=# com.android.launcher3.logger.LauncherAtom$ContainerInfo@19f79dd targetComponent=ComponentInfo{com.whatsapp/com.whatsapp.Main} screen=-1 cell(-1,-1) span(1,1) minSpan(1,1) rank=0 user=UserHandle{0} title=WhatsApp), WorkspaceItemInfo(id=-1 type=APP container=# com.android.launcher3.logger.LauncherAtom$ContainerInfo@19f79dd targetComponent=ComponentInfo{com.google.android.gm/com.google.android.gm.ConversationListActivityGmail} screen=-1 cell(-1,-1) span(1,1) minSpan(1,1) rank=0 user=UserHandle{0} title=Gmail), WorkspaceItemInfo(id=-1 type=APP container=# com.android.launcher3.logger.LauncherAtom$ContainerInfo@19f79dd targetComponent=ComponentInfo{com.instagram.android/com.instagram.android.activity.MainTabActivity} screen=-1 cell(-1,-1) span(1,1) minSpan(1,1) rank=0 user=UserHandle{0} title=Instagram), WorkspaceItemInfo(id=-1 type=APP container=# com.android.launcher3.logger.LauncherAtom$ContainerInfo@19f79dd targetComponent=ComponentInfo{com.spotify.music/com.spotify.music.MainActivity} screen=-1 cell(-1,-1) span(1,1) minSpan(1,1) rank=0 user=UserHandle{0} title=Spotify), WorkspaceItemInfo(id=-1 type=APP container=# com.android.launcher3.logger.LauncherAtom$ContainerInfo@19f79dd targetComponent=ComponentInfo{com.android.vending/com.android.vending.AssetBrowserActivity} screen=-1 cell(-1,-1) span(1,1) minSpan(1,1) rank=0 user=UserHandle{0} title=Play Store), WorkspaceItemInfo(id=-1 type=APP container=# com.android.launcher3.logger.LauncherAtom$ContainerInfo@19f79dd targetComponent=ComponentInfo{com.google.android.youtube/com.google.android.youtube.app.honeycomb.Shell$HomeActivity} screen=-1 cell(-1,-1) span(1,1) minSpan(1,1) rank=0 user=UserHandle{0} title=YouTube), WorkspaceItemInfo(id=-1 type=APP container=# com.android.launcher3.logger.LauncherAtom$ContainerInfo@19f79dd targetComponent=ComponentInfo{com.grofers.customerapp.lit/com.grofers.customerapp.DEFAULT} screen=-1 cell(-1,-1) span(1,1) minSpan(1,1) rank=0 user=UserHandle{0} title=Blinkit Lit), WorkspaceItemInfo(id=-1 type=APP container=# com.android.launcher3.logger.LauncherAtom$ContainerInfo@19f79dd targetComponent=ComponentInfo{com.trello/com.trello.home.HomeActivity} screen=-1 cell(-1,-1) span(1,1) minSpan(1,1) rank=0 user=UserHandle{0} title=Trello), WorkspaceItemInfo(id=-1 type=APP container=# com.android.launcher3.logger.LauncherAtom$ContainerInfo@19f79dd targetComponent=ComponentInfo{me.rainbow/me.rainbow.MainActivityog} screen=-1 cell(-1,-1) span(1,1) minSpan(1,1) rank=0 user=UserHandle{0} title=Rainbow), WorkspaceItemInfo(id=-1 type=APP container=# com.android.launcher3.logger.LauncherAtom$ContainerInfo@19f79dd targetComponent=ComponentInfo{com.google.android.apps.nbu.paisa.user/com.google.nbu.paisa.flutter.gpay.app.LauncherActivity} screen=-1 cell(-1,-1) span(1,1) minSpan(1,1) rank=0 user=UserHandle{0} title=GPay), WorkspaceItemInfo(id=-1 type=APP container=# com.android.launcher3.logger.LauncherAtom$ContainerInfo@19f79dd targetComponent=ComponentInfo{in.swiggy.android/in.swiggy.android.activities.HomeActivity} screen=-1 cell(-1,-1) span(1,1) minSpan(1,1) rank=0 user=UserH
2024-08-22 20:53:22.208  1724-27356 CoreBackPreview         pid-1724                             D  Window{6fd7da1 u0 Splash Screen com.cupoc.omnid EXITING}: Setting back callback null
2024-08-22 20:53:22.209  1724-27338 InputManager-JNI        pid-1724                             W  Input channel object '6fd7da1 Splash Screen com.cupoc.omnid (client)' was disposed without first being removed with the input manager!
2024-08-22 20:53:22.288  1724-27356 WindowManager           pid-1724                             W  Exception thrown during dispatchAppVisibility Window{ec94567 u0 com.cupoc.omnid/com.cupoc.omnid.MainActivity EXITING}
                                                                                                    android.os.DeadObjectException
                                                                                                    	at android.os.BinderProxy.transactNative(Native Method)
                                                                                                    	at android.os.BinderProxy.transact(BinderProxy.java:586)
                                                                                                    	at android.view.IWindow$Stub$Proxy.dispatchAppVisibility(IWindow.java:552)
                                                                                                    	at com.android.server.wm.WindowState.sendAppVisibilityToClients(WindowState.java:3217)
                                                                                                    	at com.android.server.wm.WindowContainer.sendAppVisibilityToClients(WindowContainer.java:1293)
                                                                                                    	at com.android.server.wm.WindowToken.setClientVisible(WindowToken.java:403)
                                                                                                    	at com.android.server.wm.ActivityRecord.setClientVisible(ActivityRecord.java:7100)
                                                                                                    	at com.android.server.wm.ActivityRecord.postApplyAnimation(ActivityRecord.java:5820)
                                                                                                    	at com.android.server.wm.ActivityRecord.commitVisibility(ActivityRecord.java:5762)
                                                                                                    	at com.android.server.wm.Transition.finishTransition(Transition.java:1257)
                                                                                                    	at com.android.server.wm.TransitionController.finishTransition(TransitionController.java:925)
                                                                                                    	at com.android.server.wm.WindowOrganizerController.finishTransition(WindowOrganizerController.java:489)
                                                                                                    	at android.window.IWindowOrganizerController$Stub.onTransact(IWindowOrganizerController.java:278)
                                                                                                    	at com.android.server.wm.WindowOrganizerController.onTransact(WindowOrganizerController.java:199)
                                                                                                    	at android.os.Binder.execTransactInternal(Binder.java:1505)
                                                                                                    	at android.os.Binder.execTransact(Binder.java:1444)

```

### Environment

  expo-env-info 1.2.0 environment info:
    System:
      OS: macOS 14.5
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 18.17.0 - ~/.nvm/versions/node/v18.17.0/bin/node
      Yarn: 1.22.19 - /opt/homebrew/bin/yarn
      npm: 10.7.0 - ~/.nvm/versions/node/v18.17.0/bin/npm
      Watchman: 2024.06.10.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.14.3 - /opt/homebrew/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.5, iOS 17.5, macOS 14.5, tvOS 17.5, visionOS 1.2, watchOS 10.5
      Android SDK:
        API Levels: 34
        Build Tools: 30.0.3, 31.0.0, 33.0.0, 33.0.1, 34.0.0, 35.0.0, 35.0.0
        System Images: android-33 | Google Play ARM 64 v8a, android-34 | Google APIs ARM 64 v8a
    IDEs:
      Android Studio: 2022.3 AI-223.8836.35.2231.11005911
      Xcode: 15.4/15F31d - /usr/bin/xcodebuild
    npmPackages:
      expo: ^51.0.30 => 51.0.30
      expo-router: 3.5.23 => 3.5.23
      react: 18.2.0 => 18.2.0
      react-dom: 18.2.0 => 18.2.0
      react-native: 0.75.2 => 0.75.2
    npmGlobalPackages:
      eas-cli: 10.2.3
    Expo Workflow: bare

### Expo Doctor Diagnostics

 "expo": "^51.0.30",
 "expo-updates": "0.25.23",

--- a/packages/expo-updates/CHANGELOG.md
+++ b/packages/expo-updates/CHANGELOG.md
@@ -15,7 +15,7 @@
 ### 🐛 Bug fixes
 
 - Fixed build error with React Native 0.75 on Android. ([#31084](https://github.com/expo/expo/pull/31084) by [@kudo](https://github.com/kudo))
-- Add different set of proguard rules when using React Native 0.75 on Android.
+- Add different set of proguard rules when using React Native 0.75 on Android. ([#31136](https://github.com/expo/expo/pull/31136) by [@alanjhughes](https://github.com/alanjhughes))
 
 ## 0.25.22 — 2024-08-07
 

--- a/packages/expo-updates/CHANGELOG.md
+++ b/packages/expo-updates/CHANGELOG.md
@@ -15,6 +15,7 @@
 ### 🐛 Bug fixes
 
 - Fixed build error with React Native 0.75 on Android. ([#31084](https://github.com/expo/expo/pull/31084) by [@kudo](https://github.com/kudo))
+- Add different set of proguard rules when using React Native 0.75 on Android.
 
 ## 0.25.22 — 2024-08-07
 

--- a/packages/expo-updates/android/build.gradle
+++ b/packages/expo-updates/android/build.gradle
@@ -41,6 +41,8 @@ def exUpdatesAndroidDelayLoadApp = getBoolStringFromPropOrEnv("EX_UPDATES_ANDROI
 def useDevClient = findProject(":expo-dev-client") != null
 
 android {
+  def rnVersion = getRNVersion()
+
   buildFeatures {
     buildConfig true
   }
@@ -72,12 +74,20 @@ android {
       pickFirst "**/libc++_shared.so"
     }
   }
+  buildTypes {
+    release {
+      if (rnVersion >= versionToNumber(0, 75, 0)) {
+        proguardFiles 'proguard-rules-75.pro'
+      } else {
+        proguardFiles 'proguard-rules.pro'
+      }
+    }
+  }
   sourceSets {
     main.assets.srcDirs += files("$projectDir/src/main/certificates".toString())
     androidTest.assets.srcDirs += files("$projectDir/src/androidTest/schemas".toString())
     androidTest.assets.srcDirs += files("$projectDir/src/androidTest/certificates".toString())
 
-    def rnVersion = getRNVersion()
     if (rnVersion >= versionToNumber(0, 75, 0)) {
       main.java.srcDirs += "src/react-native-75/main"
     } else {

--- a/packages/expo-updates/android/build.gradle
+++ b/packages/expo-updates/android/build.gradle
@@ -41,8 +41,6 @@ def exUpdatesAndroidDelayLoadApp = getBoolStringFromPropOrEnv("EX_UPDATES_ANDROI
 def useDevClient = findProject(":expo-dev-client") != null
 
 android {
-  def rnVersion = getRNVersion()
-
   buildFeatures {
     buildConfig true
   }
@@ -51,7 +49,7 @@ android {
   defaultConfig {
     versionCode 31
     versionName '0.25.23'
-    consumerProguardFiles("proguard-rules.pro")
+    consumerProguardFiles(rnVersion >= versionToNumber(0, 75, 0) ? 'proguard-rules-75.pro' : 'proguard-rules.pro')
     testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
 
     buildConfigField("boolean", "EX_UPDATES_NATIVE_DEBUG", exUpdatesNativeDebug)
@@ -74,20 +72,12 @@ android {
       pickFirst "**/libc++_shared.so"
     }
   }
-  buildTypes {
-    release {
-      if (rnVersion >= versionToNumber(0, 75, 0)) {
-        proguardFiles 'proguard-rules-75.pro'
-      } else {
-        proguardFiles 'proguard-rules.pro'
-      }
-    }
-  }
   sourceSets {
     main.assets.srcDirs += files("$projectDir/src/main/certificates".toString())
     androidTest.assets.srcDirs += files("$projectDir/src/androidTest/schemas".toString())
     androidTest.assets.srcDirs += files("$projectDir/src/androidTest/certificates".toString())
 
+    def rnVersion = getRNVersion()
     if (rnVersion >= versionToNumber(0, 75, 0)) {
       main.java.srcDirs += "src/react-native-75/main"
     } else {

--- a/packages/expo-updates/android/build.gradle
+++ b/packages/expo-updates/android/build.gradle
@@ -41,6 +41,7 @@ def exUpdatesAndroidDelayLoadApp = getBoolStringFromPropOrEnv("EX_UPDATES_ANDROI
 def useDevClient = findProject(":expo-dev-client") != null
 
 android {
+  def rnVersion = getRNVersion()
   buildFeatures {
     buildConfig true
   }
@@ -77,7 +78,6 @@ android {
     androidTest.assets.srcDirs += files("$projectDir/src/androidTest/schemas".toString())
     androidTest.assets.srcDirs += files("$projectDir/src/androidTest/certificates".toString())
 
-    def rnVersion = getRNVersion()
     if (rnVersion >= versionToNumber(0, 75, 0)) {
       main.java.srcDirs += "src/react-native-75/main"
     } else {

--- a/packages/expo-updates/android/proguard-rules-75.pro
+++ b/packages/expo-updates/android/proguard-rules-75.pro
@@ -1,0 +1,7 @@
+-keepclassmembers class com.facebook.react.ReactInstanceManager {
+  private final com.facebook.react.bridge.JSBundleLoader mBundleLoader;
+}
+
+-keepclassmembers class com.facebook.react.devsupport.ReleaseDevSupportManager {
+  private final com.facebook.react.bridge.DefaultJSExceptionHandler mDefaultJSExceptionHandler;
+}

